### PR TITLE
feat:[ADL] DTT feature

### PR DESCRIPTION
**Pull request by koktong-ong**
Implemeted DTT feature for ADL:
Created new CfgData_Feature for DTT
Created associated ACPI folder and files (DPTF)

Signed-off-by: Kobe <kok.tong.ong@intel.com>

### CODE REVIEW

File: Platform/AlderlakeBoardPkg/AcpiTables/Dsdt/Sen3Particip.asl [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+  //  Arguments: (0)
+  //    None
+  //  Return Value:
+  //    The return value is an integer that represents the critical sleep threshold tenths of degrees Kelvin.
+  //
+  Method(_HOT,0,Serialized)
+  {
+    Return(\_SB.PC00.LPCB.ITE8.SEN2.CTOK(S1HP))
+  }
+} // End SEN3 Device
```
Comment (sindhuragrandhi) [inline Platform/AlderlakeBoardPkg/AcpiTables/Dsdt/Sen3Particip.asl:385]: newline

File: Platform/AlderlakeBoardPkg/Library/FspmUpdUpdateLib/FspmUpdUpdateLib.c [hunk 2]
```diff
@@ -628,8 +628,6 @@ UpdateFspConfig (
         Fspmcfg->Ddr4OneDpc = MemCfgData->Ddr4OneDpc;
         Fspmcfg->FirstDimmBitMask = 0x0;
         Fspmcfg->FirstDimmBitMaskEcc = MemCfgData->FirstDimmBitMaskEcc;
-        Fspmcfg->PchIshEnable = 0x0;
```
Comment (sindhuragrandhi) [inline Platform/AlderlakeBoardPkg/Library/FspmUpdUpdateLib/FspmUpdUpdateLib.c:631]: Why these config are removed? How are these related to DTT?

File: Platform/AlderlakeBoardPkg/Library/FspsUpdUpdateLib/FspsUpdUpdateLib.c [hunk 3]
```diff
@@ -1218,11 +1218,7 @@ UpdateFspConfig (
         FspsConfig->VccInAuxImonSlope = 0x64;
         break;
       case PLATFORM_ID_ADL_PS_DDR5_CRB:
-        FspsConfig->AmtEnabled = 0x0;
```
Comment (sindhuragrandhi) [inline Platform/AlderlakeBoardPkg/Library/FspsUpdUpdateLib/FspsUpdUpdateLib.c:1221]: Lines 1221 till 1225 - Why are these config is removed?

File: Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/AcpiPlatform.c [hunk 4]
```diff
@@ -1079,12 +1106,6 @@ PlatformUpdateAcpiGnvs (
 
     PlatformNvs->PchM2SsdPowerEnableGpio = GPIO_VER4_S_GPP_K11;
     break;
-  case PLATFORM_ID_ADL_PS_DDR5_CRB:
```
Comment (sindhuragrandhi) [inline Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/AcpiPlatform.c:1082]: Why are these config lines removed? Is there a correlation with DTT?

File: Platform/AlderlakeBoardPkg/AcpiTables/Dptf/Dptf.asl [hunk 5]
```diff
@@ -0,0 +1,732 @@
...
+  // _QF1 (Query - Embedded Controller Query F1)
+  //
+  // Handler for EC generated SCI number F1.
+  //
+  // Arguments: (0)
+  //   None
+  // Return Value:
+  //   None
+  //
+  Method(_QF1)
```
Comment (sindhuragrandhi) [inline Platform/AlderlakeBoardPkg/AcpiTables/Dptf/Dptf.asl:578]: Please clean up from line 580 till 609

**Review by sindhuragrandhi — CHANGES_REQUESTED** [review]

File: Platform/AlderlakeBoardPkg/AcpiTables/AcpiTables.inf [hunk 6]
```diff
@@ -62,6 +62,7 @@
   Rtct/Rtct.act
   Lpit/Lpit.act
   Bert/Bert.aslc
+  Dptf/Dptf.asl
```
[the remark this answers is not in the document]
Comment (sindhuragrandhi) [reply Platform/AlderlakeBoardPkg/AcpiTables/AcpiTables.inf:65]: When we add any asl file here, they get loaded up even if Feature flag is disabled in configuration.

**Comment by randylintw** [comment]
Will it apply to all Alderlake platform? Maybe re-title ?

File: Platform/AlderlakeBoardPkg/AcpiTables/AcpiTables.inf (hunk 6 above)
[the remark this answers is not in the document]
Comment (bejeanmo) [reply Platform/AlderlakeBoardPkg/AcpiTables/AcpiTables.inf]: You can also prevent a table from being installed by returning an error status from PlatformUpdateAcpiTable(), but if it's a large table, it's good to avoid building it in.

**Comment by sindhuragrandhi** [comment]
@koktong-ong please address check patch format issues:
Code format is not valid:
 * No newline at end of file Platform/AlderlakeBoardPkg/AcpiTables/Dsdt/IteCom.asl
 * No newline at end of file Platform/AlderlakeBoardPkg/AcpiTables/Dsdt/IteHwMonAuto.asl
